### ISSUE TITLE
Probelm with deserialization of secret key struct

### ISSUE THREAD

**Issue by evgeniy-scherbina** [rust-bitcoin/rust-secp256k1#73]
At this moment this trivial example fails with such error `invalid type: sequence, expected a borrowed byte array"`
```rust
extern crate serde;
extern crate serde_json;
extern crate secp256k1;

use secp256k1::{Secp256k1, SecretKey};

fn main() {
    let sk = [1u8; 32];
    let sk = SecretKey::from_slice(&Secp256k1::new(), &sk).unwrap();

    let encoded = serde_json::to_string(&sk).unwrap();
    let _decoded: SecretKey = serde_json::from_str(&encoded).unwrap();
}
```

It seems that minor fix to implementation of Deserialize trait may fix this problem.

**Comment by apoelstra** [comment]
It looks like I need to use the `serde_bytes` crate to do this (or copy its tremendous amount of boilerplate into this library). This will require moving a second crate under the `serde` feature, which means renaming the feature, so it will have to wait for 0.13.

**Comment by jonasnick** [comment]
Should've been fixed by #107.
